src/bun.js/bindings/webcore/JSWorker.cpp

```diff
@@ -117,18 +120,33 @@ template<> JSC::EncodedJSValue JSC_HOST_CALL_ATTRIBUTES JSWorkerDOMConstructor::
     auto& vm = JSC::getVM(lexicalGlobalObject);
     auto throwScope = DECLARE_THROW_SCOPE(vm);
     auto* castedThis = jsCast<JSWorkerDOMConstructor*>(callFrame->jsCallee());
+    auto* globalObject = defaultGlobalObject(lexicalGlobalObject);
     ASSERT(castedThis);
     if (UNLIKELY(callFrame->argumentCount() < 1))
         return throwVMError(lexicalGlobalObject, throwScope, createNotEnoughArgumentsError(lexicalGlobalObject));
     auto* context = castedThis->scriptExecutionContext();
     if (UNLIKELY(!context))
         return throwConstructorScriptExecutionContextUnavailableError(*lexicalGlobalObject, throwScope, "Worker"_s);
     EnsureStillAliveScope argument0 = callFrame->uncheckedArgument(0);
-    auto scriptUrl = convert<IDLUSVString>(*lexicalGlobalObject, argument0.value());
+    String scriptUrl;
+    if (auto* url = jsDynamicCast<JSDOMURL*>(argument0.value())) {
+        scriptUrl = url->wrapped().href().string();
```

Comment:
can you check if this needs to handle invalid URLs? can they be invalid? I don't quite remember.